### PR TITLE
Upgrade primus and engine io depedencies

### PR DESCRIPTION
**Pull request by camallen**
dependency upgrades

primus now has switched from a
`client ping -> server pong connection keepalive` 
to 
`server ping -> client pong connection keepalive` 
paradigm
https://github.com/primus/primus/releases/tag/7.0.0

### CODE REVIEW

**Comment by camallen** [comment]
closing in favour of a straight upgrade of the dev dependency in #69

File: test/support/sugar_client.coffee [hunk 1]
```diff
@@ -7,42 +7,63 @@ class SugarClient
...
+    # client would write a ping message
+    # @client.write "primus::ping::#{ +new Date() }"
+    # server would respond with a pong
+    # and client would dispatch the incoming pong event
+    # @client.once 'incoming::pong', -> deferred.resolve()
+
+    # new server -> client behaviour
+    # server sends a ping
+    # client dispatches the incoming ping event
+    @client.emit 'incoming::ping', -> deferred.resolve()
```
Comment (eatyourgreens) [inline test/support/sugar_client.coffee:40]: I’m pretty sure this should be a listener, not an emitter.

Comment (eatyourgreens) [reply test/support/sugar_client.coffee:40]: `@client.once` might fix the tests.

Comment (eatyourgreens) [reply test/support/sugar_client.coffee:40]: So, my dumb solution to this is to wait for 100ms then resolve the promise. See #106.
```suggestion
    setTimeout ->
      deferred.resolve()
    , 100
```

**Pull request closed without merging**